### ISSUE TITLE
flushMicrotasks cannot be the only way to make microtasks run

### ISSUE THREAD

**Issue by zanderso** [dart-lang/fake_async#32]
Otherwise, tests may hang. See the discussion on https://github.com/flutter/flutter/issues/75155. In particular https://github.com/flutter/flutter/issues/75155#issuecomment-772394162 and https://github.com/flutter/flutter/issues/75155#issuecomment-773435635.

**Comment by pinkasey** [comment]
I think this is the issue I'm running into now -
I'm writing my first widget-test, and I get `pumpAndSettle timed out` failures on the first time I call `pumpAndSettle()`.
calling `pumpAndSettle(Duration(days: 20))` didn't help - it's stuck in an infinite loop.

I placed a breakpoint at `fake_async.dart:185`:
```
                  _microtasks.add(microtask)));
```

and I saw these two stack-traces:
![image](https://user-images.githubusercontent.com/1780369/107452345-b0f06680-6b51-11eb-9ca5-3fa82a38cce9.png)
(note `pumpAndSettle` at the bottom)

![image](https://user-images.githubusercontent.com/1780369/107452789-8521b080-6b52-11eb-87c9-2ed84f3ea283.png)
(note `flushMicrotasks` at the bottom)

So I think that every iteration of `pump` adds a micro-task when it's done, so `pumpAndSettle` never finishes.
